### PR TITLE
add Content Panels for Musicvideos in SearchEngine

### PR DESCRIPTION
**Pull request by marduklev**
- find titles by searchstring
- find artist by search string

### CODE REVIEW

File: 1080i/Custom_1138_Search.xml [hunk 1]
```diff
@@ -240,7 +240,50 @@
...
+                    <param name="label2" value="ListItem.Label" />
+                    <param name="visible" value="Integer.IsGreater(Container(5015).NumItems,0) | Container(5015).IsUpdating" />
+                    <animation effect="fade" start="100" end="0" condition="!Integer.IsGreater(Container(5015).NumItems,0) | Container(5015).IsUpdating">Conditional</animation>
+                    <content target="videos">videodb://musicvideos/titles/?xsp=%7b%22order%22%3a%7b%22direction%22%3a%22ascending%22%2c%22ignorefolders%22%3a0%2c%22method%22%3a%22sorttitle%22%7d%2c%22rules%22%3a%7b%22and%22%3a%5b%7b%22field%22%3a%22title%22%2c%22operator%22%3a%22contains%22%2c%22value%22%3a%5b%22$INFO[Skin.String(SearchTerm)]%22%5d%7d%5d%7d%2c%22type%22%3a%22musicvideos%22%7d</content>
+                </include>
+				<!-- MUSICVIDEOS by SearchString in Artist -->
+                <include content="View_Cast_Toplabel">
+                    <param name="groupid" value="7516" />
+                    <param name="id" value="5016" />
+                    <param name="label" value="$INFO[Container(5016).NumItems,$LOCALIZE[20389] by (,)]" />
```
Comment (jurialmunkey) [inline 1080i/Custom_1138_Search.xml]: Can you localize the "by" on this line?

**Review by jurialmunkey — CHANGES_REQUESTED** [review]
Looks good! Can you just localize the "by" on line 269 and then I will merge. Thanks 😃

**Comment by jurialmunkey** [comment]
Hi. Just wondering if you were going to make the above requested change?

**Comment by marduklev** [comment]
Sorry for late response, will check next monday.
msgctxt "#1443"
msgid "of"
msgstr "von"


I need also to check the artwork image

**Comment by jurialmunkey** [comment]
Not a problem. No need to rush - was just making sure you hadn't missed my comments.

**Comment by marduklev** [comment]
ok, unsure if it'll cause wrong it'll cause "grammar issues" if it'll use that 

 - $INFO[Skin.String(SearchTerm),$LOCALIZE[20391] - $LOCALIZE[133] $LOCALIZE[21400]  &quot; , &quot; ] $INFO[Container(5001).NumItems, (,)]

music videos - artist contains "searchstring" (numitems)

or if i should do pr , with a predefined label in language strings(.po)
 - $INFO[Skin.String(SearchTerm),$LOCALIZE[20389] $LOCALIZE[31432] &quot;[CAPITALIZE],[/CAPITALIZE]&quot;]$INFO[Container(5013).NumItems, (,)]
 
 = Music viedeos - "by Artist named" 'Searchstring' 

 - #: /1080i/Custom_1138_Search.xml
 - msgctxt "#31432"
 - msgid "By Artist Named"
 - msgstr ""

**Comment by jurialmunkey** [comment]
Grammatical issues might be a problem, so I've been trying to keep it as simple as possible. 
e.g. For Albums by Artist I used the format: "Albums by 'Artist'"
The "Albums by" is localized as one string.

Maybe this would be best:
Music Videos by 'SearchString'
"Music Videos by" as one localized string.

That way, if it is a complete string, then there is at least some control in the translation for phrasing.

**Comment by marduklev** [comment]
- i think its done, at least , you should just check if you really want ListItem.Icon, instead of an variable for fanart

# e.g.
```
<value condition="!String.IsEmpty(ListItem.Art(artist.landscape))">$INFO[ListItem.Art(artist.landscape)]</value>
	<value condition="!String.IsEmpty(ListItem.Art(albumartist.landscape))">$INFO[ListItem.Art(albumartist.landscape)]</value>
	<value condition="!String.IsEmpty(ListItem.Art(landscape))">$INFO[ListItem.Art(landscape)]</value>
	<value condition="!String.IsEmpty(ListItem.Art(artist.fanart))">$INFO[ListItem.Art(artist.fanart)]</value>
	<value condition="!String.IsEmpty(ListItem.Art(albumartist.landscape))">$INFO[ListItem.Art(albumartist.fanart)]</value>
	<value condition="!String.IsEmpty(ListItem.Art(fanart))">$INFO[ListItem.Art(fanart)]</value>
	<value>$INFO[ListItem.Icon]</value>
```

**Comment by jurialmunkey** [comment]
I made a few small changes, but everything has been merged.
Changed artwork to square as musicvideos default scraper scrapes album covers as ListItem.Icon and there doesn't seem to be anyway to get the actual video thumbnail (which is what I would actually prefer).
Changed the "by artist named" string to be one string "Music Videos by Artist Named" rather than two combined.

**Pull request merged**